plugins/pulp_python/plugins/models.py

```diff
@@ -3,23 +3,58 @@
 import re
 import tarfile
 
+from mongoengine import StringField
+from pulp.server.db.model import FileContentUnit
+
 from pulp_python.common import constants
 
 
 DEFAULT_CHECKSUM_TYPE = 'sha512'
 
 
-class Package(object):
+# These are required to be in the PKG-INFO file.
+REQUIRED_ATTRS = ('name', 'version', 'summary', 'home_page', 'author', 'author_email', 'license',
+                  'description', 'platform')
+
+
+class Package(FileContentUnit):
     """
     This class represents a Python package.
     """
 
-    TYPE = constants.PACKAGE_TYPE_ID
-    # The full list of supported attributes. Attributes beginning with underscore are specific to
-    # this module and are not found in PKG-INFO.
-    _ATTRS = ('name', 'version', 'summary', 'home_page', 'author', 'author_email', 'license',
-              'description', 'platform', '_filename', '_checksum', '_checksum_type',
-              '_metadata_file')
+    # unit key
+    name = StringField(required=True)
+    version = StringField(required=True)
+
+    author = StringField()
+    author_email = StringField()
+    description = StringField()
+    home_page = StringField()
+    license = StringField()
+    platform = StringField()
+    summary = StringField()
+
+    _checksum = StringField()
+    _checksum_type = StringField(default=DEFAULT_CHECKSUM_TYPE)
```

Comment:
I also kind of regret making the checksum be two fields instead of just using a notation like algo:hash and storing that in one field. This is one thing I do like about how Docker did it!